extension/src/plots/model/index.ts

```diff
@@ -101,15 +102,23 @@ export class PlotsModel extends ModelWithPersistence {
 
     this.setMetricOrder()
 
-    return this.removeStaleData()
+    return this.removeStaleData(branchRevisions, revisions)
   }
 
-  public async transformAndSetPlots(data: PlotsOutput, revs: string[]) {
-    const cliIdToLabel = this.getCLIIdToLabel()
+  public async transformAndSetPlots(
+    data: PlotsOutput,
+    newRevs: string[],
+    selectedRevs: SelectedExperimentWithColor[]
+  ) {
+    if (data.error) {
+      return
+    }
+
+    const cliIdToLabel = this.getCLIIdToLabel(selectedRevs)
 
     this.fetchedRevs = new Set([
       ...this.fetchedRevs,
-      ...revs.map(rev => cliIdToLabel[rev])
+      ...newRevs.map(rev => cliIdToLabel[rev])
     ])
```

Comment:
[F] I think we can expand `fetchedRevs` to include state information about each revision. We can then display this information on the ribbon. The 4 states would be:

- loading
- loaded with data
- loaded no data
- loaded with errors

We could potentially combine 3 & 4.

Reply:
WDYT?